pkg/azuredisk/controllerserver.go

```diff
@@ -201,6 +210,15 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 			if strings.EqualFold(v, trueValue) {
 				enableBursting = to.BoolPtr(true)
 			}
+		case userAgentField:
+			nonUserAgentCloud = d.cloud
+			newUserAgent := v
+			d.cloud, err = SetupNewUserAgentCloud(newUserAgent, d.cloud)
```

Comment:
We don't want to change the global instance of the cloud provider as we only want the custom user agent string used for the duration of this `CreateVolume` call and not for any others. Use a local variable initialized to `d.cloud` for the `CreateVolume` call and override the value here instead.

Reply:
actually in Line 219, it already changes the global instance. It's hard to set multiple cloud provider instances for multiple  userAgents unless we set a big lock in CreateVolume to block other create volume activities until it succeeds, while set a lock would bring performance issue.